lib/templates/lab2/lab2.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:group_button/group_button.dart';
+
+void main() => runApp(const Lab1());
+
+class Lab1 extends StatelessWidget {  
+  const Lab1({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: LAb1HomePage(),
+    );
+  }
+}
+
+class LAb1HomePage extends StatelessWidget {
+  const LAb1HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        leading: const Icon(Icons.stars),
+        title: const Text('Lab 2'),
+        backgroundColor: Colors.teal,
+        elevation: 4,
+      ),
+      body: myWidget(),
+    );
+  }
+}
+
+Widget myWidget() {
+  
+  Book book = Book("Atlas Shrugged", "Ayn Rand", 1364);
+
+  book.bookDetails();
+
+  FictionBook fictionBook = FictionBook("The Lord of the Rings", "John Ronald Reuel Tolkien", 1207, "Fantasy");
+
+  fictionBook.bookDetails();
+
+  User user = User(name: "Scott Pilgrim", age: 23, email: "scottpilgrim@example.boom");
+
+  user.printUserDetails();
+
+  return Container();
+}
+
+class Book {
+  String title;
+  String author;
+  int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+
+  bookDetails() {
+    print("Title: $title");
+    print("Author: $author");
+    print("Number of Pages: $numberOfPages");
+  }
+}
+
+class FictionBook extends Book {
+  String genre;
+
+  FictionBook(String title, String author, int numberOfPages, this.genre)
+      : super(title, author, numberOfPages);
+
+  @override
+  bookDetails() {
+    super.bookDetails();
+    print("Title: $title");
+    print("Author: $author");
+    print("Number of Pages: $numberOfPages");
+    print("Genre: $genre");
+  }
+}
+
+mixin JsonSerializable on Book {
```

Comment:
<img width="121" alt="Screenshot 2024-06-11 at 06 55 50" src="https://github.com/timur-harin/Sum24FlutterCourse/assets/64702874/db8696f1-6e2c-4c67-ada0-21ee4b79e684">

Reply:
Could not run it